### PR TITLE
integrate mail config to api-server roles

### PR DESCRIPTION
**Pull request by Pilzington**
implements: https://github.com/metal-stack-cloud/api-server/pull/239
I tried to prepare the PR for integrating our Mail Setup into the deployment. I am not sure if it is correct like this.

### CODE REVIEW

File: roles/api-server/files/api-server/templates/jobs.yaml [hunk 1]
```diff
@@ -35,3 +35,40 @@ spec:
...
+            - name: MJ_APIKEY_PRIVATE
+              value: "{{ .Values.mail.mjApiKeyPublic }}"
+          args:
+            - stripe-jobs check-for-coupon-usage
+            - --log-level={{ .Values.api.logLevel }}
+            - --stripe-secret-key={{ .Values.api.stripe.secretKey }}
+            - --mail-provider={{ .Values.mail.mailProvider }}
+            - --mail-config-path=/mail/mail-config.yaml
+            - --smtp-from-address={{ .Values.mail.smtpFromAddress }}
+            - --smtp-from-name={{ .Values.mail.smtpFromName }
```
Comment (vknabel) [inline roles/api-server/files/api-server/templates/jobs.yaml:72]:
```suggestion
            - --smtp-from-name={{ .Values.mail.smtpFromName }}
```

**Comment by vknabel** [comment]
Sadly I got an error while running this:

```
cloud    | TASK [ansible-common/roles/helm-chart : Deploy helm chart (api-server)] ********
cloud    | fatal: [localhost]: FAILED! => changed=true 
cloud    |   cmd:
cloud    |   - helm
cloud    |   - upgrade
cloud    |   - --install
cloud    |   - --namespace
cloud    |   - metal-stack-cloud
cloud    |   - -f
cloud    |   - values.yaml
cloud    |   - --wait
cloud    |   - api-server
cloud    |   - ./api-server
cloud    |   delta: '0:00:00.051263'
cloud    |   end: '2023-04-21 13:28:37.778052'
cloud    |   msg: non-zero return code
cloud    |   rc: 1
cloud    |   start: '2023-04-21 13:28:37.726789'
cloud    |   stderr: 'Error: UPGRADE FAILED: YAML parse error on api-server/templates/deployment.yaml: error converting YAML to JSON: yaml: line 31: did not find expected key'
cloud    |   stderr_lines: <omitted>
cloud    |   stdout: ''
cloud    |   stdout_lines: <omitted>
cloud    | 
cloud    | PLAY RECAP *********************************************************************
cloud    | localhost                  : ok=7    changed=3    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
cloud    | 
cloud exited with code 2
```

Will have a look into this on monday

File: roles/api-server/files/api-server/templates/deployment.yaml [hunk 2]
```diff
@@ -27,6 +27,12 @@ spec:
             value: "{{  .Values.frontend.url }}"
           - name: USER_ADMITTANCE
             value: "{{ .Values.api.userAdmittance }}"
+{{- if eq .Values.mail.mailProvider "mailjet" }}
+          - name: MJ_APIKEY_PUBLIC
+            value: "{{ .Values.mail.mjApiKeyPublic }}"
+          - name: MJ_APIKEY_PRIVATE
+            value: "{{ .Values.mail.mjApiKeyPublic }}"
+{{- end -}}
```
Comment (vknabel) [inline roles/api-server/files/api-server/templates/deployment.yaml:35]: The trailing trimming `-` produced this error: https://github.com/metal-stack-cloud/ansible-roles/pull/16#issuecomment-1517837701

```suggestion
{{- end }}
```

Comment (majst01) [reply roles/api-server/files/api-server/templates/deployment.yaml:35]: @vknabel please push your fixes to this PR that we can get this out and released

File: roles/api-server/defaults/main/main.yaml [hunk 3]
```diff
@@ -44,3 +44,17 @@ api_server_hpa_max: 5
 api_server_hpa_cpu_percentage: 70
 
 api_server_resources:
+
+api_server_mail_provider: none
```
Comment (majst01) [inline roles/api-server/defaults/main/main.yaml:48]: Documentation for these properties would be nice

**Review by majst01 — APPROVED** [review]

File: roles/api-server/defaults/main/main.yaml (hunk 3 above)
Comment (vknabel) [reply roles/api-server/defaults/main/main.yaml:48]: Created an issue to not forget this topic: https://github.com/metal-stack-cloud/ansible-roles/issues/17

**Pull request merged**